src/python_testing/TestReadSubscribeAceExistenceErrors.py

```diff
@@ -0,0 +1,1070 @@
+                                                expected_count=1
+                                                )
+
+        self.assert_expected_event_status_count(status=Status.UnsupportedCluster,
+                                                events=read_step4c,
+                                                expected_count=1
+                                                )
+
+    @async_test_body
+    async def test_subscribe_access_existence(self):
+
+        aclAttr = Clusters.AccessControl.Attributes.Acl
+        validAclAttrPath = (ROOT_NODE_ENDPOINT_ID, aclAttr)
+
+        vendorIdAttr = Clusters.BasicInformation.Attributes.VendorID
+        validBasicInformationVendorIDPath = (ROOT_NODE_ENDPOINT_ID, vendorIdAttr)
+
+        UnsupportedEndpointAclAttrPath = (ROOT_NODE_ENDPOINT_ID + 80, aclAttr)
+        UnsupportedClusterAclAttrPath = (ROOT_NODE_ENDPOINT_ID + 1, aclAttr)
+        #######################################    Step 1: Subscribe to Single Attribute on Valid Path with No Access    ################################################
+        #
+        #
+        # Attribute Paths:
+        #   1.Valid Attribute Path (NO ACCESS)
+        #
+        # Expected Responses:
+        #   1. INVALID_ACTION
+        #
+        self.print_step(1, "Subscribe Request to valid Concrete Attribute Path; No privileges granted to cluster under test")
+
+        await self.restore_acls_to_th1_only()
+
+        with asserts.assert_raises(ChipStackError) as cm:
+            await self.TH2.ReadAttribute(
+                nodeid=self.dut_node_id,
+                attributes=[validAclAttrPath],
+                keepSubscriptions=False,
+                reportInterval=(1, 5),
+                autoResubscribe=False
+            )
+        asserts.assert_equal(cm.exception.err, INVALID_ACTION_ERROR_CODE,
+                             "Expected Invalid_Action since all paths are discarded")
+
+        #######################################    Step 2: Subscribe to Single Attribute on Unsupported Cluster with No Access   ################################################
+        #
+        #
+        # Attribute Paths:
+        #   1.Attribute Path on an Unsupported Cluster (NO ACCESS)
+        #
+        # Expected Responses:
+        #   1. INVALID_ACTION
+        #
+        self.print_step(2, "Subscribe Request to Attribute with Unsupported Cluster; No privileges granted to cluster under test")
+
+        await self.restore_acls_to_th1_only()
+
+        with asserts.assert_raises(ChipStackError) as cm:
+            await self.TH2.ReadAttribute(
+                nodeid=self.dut_node_id,
+                attributes=[UnsupportedClusterAclAttrPath],
+                keepSubscriptions=False,
+                reportInterval=(1, 5),
+                autoResubscribe=False
+            )
+
+        # We receive an Invalid Action rather than Unsupported Access, since no Report Data was sent
```

Comment:
Is this expected per the spec? This seems like it also leaks existence. Or do non-existance and no access both return INVALID_ACTION?